llpc/lower/llpcSpirvLowerRayTracing.cpp

```diff
@@ -109,66 +109,96 @@ static unsigned TraceParamsTySize[] = {
     9, // 16, HitTriangleVertexPositions
 };
 
-// ====================================================================================================================
-// Get payload idx for TraceRayKHR instruction.
-unsigned getTraceRayParamPayloadIdx(void) {
-  return TraceRayParam::Payload;
-}
-
 // =====================================================================================================================
 SpirvLowerRayTracing::SpirvLowerRayTracing() : SpirvLowerRayQuery(false) {
 }
 
 // =====================================================================================================================
-// Create function and link traceRays traceRaysinline function
+// Process a trace ray call by creating (or get if created) an implementation function and replace the call to it.
 //
-// @param func : The trace function to create
-// @param opcode : OpTraceRayKHR or OpTraceNV to process
-template <> void SpirvLowerRayTracing::createRayTracingFunc<OpTraceRayKHR>(Function *func, unsigned opcode) {
-  // Empty function
-  assert(func->size() == 0);
-  func->setName(func->getName() + m_module->getName());
-  auto rayTracingContext = static_cast<RayTracingContext *>(m_context->getPipelineContext());
-  bool indirect = rayTracingContext->getIndirectStageMask() & ShaderStageComputeBit;
+// @param inst : The original call instruction
+void SpirvLowerRayTracing::processTraceRayCall(BaseTraceRayOp *inst) {
+  std::string mangledName(inst->getCalledFunction()->getName());
```

Comment:
Can be replaced with `std::string mangledName = inst->getCalledFunction()->getName().str() + ".impl";`